### ISSUE TITLE
jenkins pipeline error from podman in podman: error in copier subprocess: error changing to intended-new-root directory \"/var/lib/containers/storage/overlay/5d...f1/merged\": chdir /var/lib/containers/storage/overlay/5d...f1/merged: invalid argument

### ISSUE THREAD

**Issue by zzguang520** [containers/podman#10100]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug
I thinks it is bug

/kind feature

**Description**

I have create pipeline in jenkins for auto build image by podman, part of the pipeline code:
```
	podTemplate(
		label: 'image-build', cloud: 'kubernetes', containers: [
			containerTemplate(
				name: 'build-tools',
				image: 'quay.io/containers/podman:latest', 
				command: 'cat', 
				privileged: true,
				ttyEnabled: true
			)
		],
		volumes: [
			hostPathVolume(mountPath: '/var/lib/containers', hostPath: '/var/lib/containers')
		])
```
but have error when used podman to build images by Dockerfile:
```
TASK [build test images ] ***
task path: /home/jenkins/agent/workspace/test-auto-deploy/auto_build.yml:1
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to build image harbor.test.com:7443/test-nginx:v1.0.0_20210420:  Error: error building at STEP \"WORKDIR /usr/share/nginx\": error in copier subprocess: error changing to intended-new-root directory \"/var/lib/containers/storage/overlay/5dc9f8ddcc4f1a0ebe6322ea614d64581f4724c0bd6fd9d9d195976a1aafa3f1/merged\": chdir /var/lib/containers/storage/overlay/5dc9f8ddcc4f1a0ebe6322ea614d64581f4724c0bd6fd9d9d195976a1aafa3f1/merged: invalid argument\n"}
```
```
cat auto_build.yml
- name: build test images
  containers.podman.podman_image:
    name: "{{Registry_Host}}/{{Profile_Name}}/{{NGINX_Service_Name}}"
    tag: "{{NewVersion}}"
    push: yes
    username: "{{Login_Username}}"
    password: "{{Login_Password}}"
    path: "{{Code_Folder}}"
    build: 
      format: docker
    push_args:
      compress: yes
      dest: "{{Registry_Host}}/{{Profile_Name}}/{{NGINX_Service_Name}}:{{NewVersion}}"
    validate_certs: no
```
```
cat Dockerfile
FROM nginx
WORKDIR /usr/share/nginx
RUN echo '<h1>hello, docker!</h1>' > /usr/share/nginx/html/index.html
```
I can confirm that the Dockerfile and auto_build.yml was works

**Steps to reproduce the issue:**

1.create pipeline in jenkins include podTemplate
2.install ansible etc. in pod
3.part of pipeline code for stage
```
            stage('auto-build-images') {
		    try {
			    if (fileExists ('ansible/auto_build.yml')) {
				    sh "ansible-playbook -vv ansible/auto_build.yml"
			    }
		    } catch (err) {
				    throw err
		    }
            }
```

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**
jenkins deploy in k8s of 1.20
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:20:00Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
**Output of `podman version`:**
```
Version:            1.6.4
RemoteAPI Version:  1
Go Version:         go1.12.12
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**
```
debug:
  compiler: gc
  git commit: ""
  go version: go1.12.12
  podman version: 1.6.4
host:
  BuildahVersion: 1.11.5
  CgroupVersion: v1
  Conmon:
    package: conmon-2.0.8-1.el7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.8, commit: f85c8b1ce77b73bcd48b2d802396321217008762'
  Distribution:
    distribution: '"rhel"'
    version: "7.9"
  MemFree: 8863678464
  MemTotal: 16654729216
  OCIRuntime:
    name: runc
    package: containerd.io-1.4.4-3.1.el7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc93
      commit: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
      spec: 1.0.2-dev
      go: go1.13.15
      libseccomp: 2.3.1
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 16
  eventlogger: journald
  hostname: master1.ioccx.ibm.com
  kernel: 3.10.0-1160.15.2.el7.x86_64
  os: linux
  rootless: false
  uptime: 1012h 39m 44.62s (Approximately 42.17 days)
registries:
  blocked: null
  insecure: null
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions: {}
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 1
  RunRoot: /var/run/containers/storage
  VolumePath: /var/lib/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**
check in master and node for `rpm -q` 
```
podman-1.6.4-29.el7_9.x86_64 
```
podman info in quay.io/containers/podman:latest
```
host:
  arch: amd64
  buildahVersion: 1.20.0
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.27-2.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.27, commit: '
  cpus: 16
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: file
  hostname: ioccx-build-wwmrh-9z4gt
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 3.10.0-1160.15.2.el7.x86_64
  linkmode: dynamic
  memFree: 6415421440
  memTotal: 16654729216
  ociRuntime:
    name: crun
    package: crun-0.19-1.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.19
      commit: e67a75672412975916dac6b87f8346f870e4b99a
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    selinuxEnabled: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 1011h 32m 0.16s (Approximately 42.12 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /var/lib/shared
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.5.0-1.fc33.x86_64
      Version: |-
        fusermount3 version: 3.9.3
        fuse-overlayfs: version 1.5
        FUSE library version 3.9.3
        using FUSE kernel interface version 7.31
    overlay.mountopt: nodev,fsync=0
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 1
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.1.0
  Built: 1618238356
  BuiltTime: Mon Apr 12 14:39:16 2021
  GitCommit: ""
  GoVersion: go1.15.8
  OsArch: linux/amd64
  Version: 3.1.0
```
**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by mheon** [comment]
Please try a more recent Podman version. v3.0 and up featured an extensive rework of copying, for example.

Also, your `podman info` and package version don't match at all - one is from EL7, the other from F33?

**Comment by zzguang520** [comment]
@mheon I have tried update to the latest version for podman from EL7,`podman version`
```
Version:      3.0.1
API Version:  3.0.0
Go Version:   go1.15.5
Built:        Mon Mar  1 22:30:40 2021
OS/Arch:      linux/amd64
[root@master1 ~]# podman info
host:
  arch: amd64
  buildahVersion: 1.19.4
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.27-1.el7.1.1.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.27, commit: '
  cpus: 16
  distribution:
    distribution: '"rhel"'
    version: "7.9"
  eventLogger: journald
  hostname: master1.ioccx.ibm.com
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 3.10.0-1160.15.2.el7.x86_64
  linkmode: dynamic
  memFree: 8859811840
  memTotal: 16654729216
  ociRuntime:
    name: runc
    package: containerd.io-1.4.4-3.1.el7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc93
      commit: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
      spec: 1.0.2-dev
      go: go1.13.15
      libseccomp: 2.3.1
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    selinuxEnabled: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 1027h 5m 52.73s (Approximately 42.79 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 1
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.0.0
  Built: 1614609040
  BuiltTime: Mon Mar  1 22:30:40 2021
  GitCommit: ""
  GoVersion: go1.15.5
  OsArch: linux/amd64
  Version: 3.0.1
```
But still this error from pipeline running

**Comment by zzguang520** [comment]
`rpm -q podman`
```podman-3.0.1-2.el7.3.1.x86_64```

**Comment by mheon** [comment]
@rhatdan You ever seen this one before? Looks like podman-in-container, but an EINVAL seems wierd (chdir manpage doesn't even document it).

I would suspect a permissions issue - entirely possible mounting filesystems is blocked by the other container - but that doesn't seem like it would cause an EINVAL, unless mount is set to return cleanly with Seccomp despite being denied?

**Comment by rhatdan** [comment]
The problem is most likely caused by some new fields specified on the mount point that RHEL7 kernel does not understand.

zzguang520  Could you check the mount options in storage.conf?

**Comment by zzguang520** [comment]
`cat /etc/containers/storage.conf | grep -v -E "^#|^$"`
```
[storage]
driver = "overlay"
runroot = "/run/containers/storage"
graphroot = "/var/lib/containers/storage"
[storage.options]
additionalimagestores = [
]
[storage.options.overlay]
mountopt = "nodev"
[storage.options.thinpool]
```

**Comment by rhatdan** [comment]
@zzguang520 Were you able to get any further?

**Comment by rhatdan** [comment]
@giuseppe Any ideas?  I would think there might be a flag in overlay mounts that are using a new option not understood by RHEL7 kernel?

**Comment by giuseppe** [comment]
could be.

But the error seems to be returned after a `chdir` and it seems strange that the underlying file system makes any difference.  Looking at how chdir works, it seems strange to fail with `EINVAL`.

@zzguang520 Is there a way we could strace the nested podman and see exactly what syscall is failing and what arguments were passed to it?

**Comment by stevegt** [comment]
I'm hitting the same EINVAL error with buildah 1.21.0 on ubuntu 20.04 when running `buildah run alpine-working-container sh` -- a simpler test like that might make troubleshooting easier for OP.  (And no, I don't know yet what's causing it either.)

**Comment by stevegt** [comment]
The OP's kernel:  ` kernel: 3.10.0-1160.15.2.el7.x86_64`
My kernel was: `4.8.0-58-generic`

Problem went away for me when I upgraded to kernel 5.4.0.  Turns out it's possible, starting with an older image and upgrading packages selectively, to get the host image into a state where the buildah package installation will succeed, but the system does not have all the prerequisites.   (In my case I was using kubic.)

I wasn't able to prove that the 4.8.0 kernel itself was the issue; it could be that, or it could be related to one of the other packages which was replaced by the 5.4.0 upgrade.

tl;dr  You need a recent kernel along with its prereqs.

**Comment by stevegt** [comment]
Just now found this in https://github.com/containers/fuse-overlayfs/blob/master/README.md:

> Also, please note that, when using fuse-overlayfs from a user namespace (for example, when using rootless podman) a Linux Kernel > v4.18.0 is required.

So yeah, the 4.8.0 kernel I was running isn't supposed to work, let alone the 3.10.0 kernel in OP's case.  

I've submitted PR containers/buildah#3295 to update the buildah docs.

The podman installation instructions don't mention a minimum kernel version at all -- that should probably be fixed too:  https://podman.io/getting-started/installation

**Comment by vrothberg** [comment]
@rhatdan @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
This issue with kernel versions is it ignores kernels like RHEL/CentOS where fixes are back ported to certain kernels.

**Comment by rhatdan** [comment]
I believe the docs are updated, reopen if I am mistaken.
